runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java

```diff
@@ -192,4 +200,21 @@ private Specification<Process> isCreatedByUser() {
     private Specification<Process> isPublic() {
         return (root, query, criteriaBuilder) -> criteriaBuilder.isTrue(root.get(Process_.isPublic));
     }
+
+    public List<ProcessDTO> filterGuestProcessesByUserRole(List<ProcessDTO> processes, String userRoles) {
+        boolean hasRequesterRoleAdmin = userRoles.contains(AuthoritiesConstants.ADMIN);
+        List<Long> guestIds = this.userService.getAllGuestIds();
+        List<ProcessDTO> filteredProcesses = new ArrayList<>();
+
+        if(!hasRequesterRoleAdmin) {
+            for (ProcessDTO process : processes) {
+                if(!guestIds.contains(process.getCreatedBy().getId())) {
+                    filteredProcesses.add(process);
+                }
+            }
+            return filteredProcesses;
+        }
```

Comment:
all of this you can replace with a single stream chain

Reply:
DONE